homeassistant/components/statistics/sensor.py

```diff
@@ -1062,7 +1067,7 @@ async def _initialize_from_database(self) -> None:
             self._fetch_states_from_database
         ):
             for state in reversed(states):
-                self._add_state_to_queue(state)
+                self._add_state_to_queue(state, state.last_reported_timestamp)
```

Comment:
The state read from the database will not be changed, which means it's safe to use the `last_reported_timestamp`